types.ts

```diff
@@ -27,10 +27,11 @@ export type WeeklyArticleHistory = WeeklyArticleLog[];
 
 export type Targeting = {
     contentType: string;
-    sectionName: string;
+    sectionName?: string;
```

Comment:
Splitting the two sounds like a good decision to me.

If the 'canonical' types are in support-dotcom-components, could we just import them from there?

Reply:
It looks like this `Targeting` definition is only used by the epic (`SlotBodyEnd`) - is that right?
What does `automat-client` give us?

Reply:
Yeah, it's enforced (not particularly explicitly) by the use of `getBodyEnd` from automat-client, which uses the `Metadata` type which includes `Targeting` and `Tracking`.

Reply:
From testing, I don't think `ReaderRevenueBanner` has any type control at all (you can change any of the properties without a complaint from Typescript) - I think ideally it would use the types from elsewhere though - the problem is at the moment the types it includes aren't included in the definitions from automat-client. From what Tom W says, perhaps there are more canonical types in support-dotcom-components?

Reply:
> What does automat-client give us?

Reminding myself by looking over the repo. It looks like the main things are:

* type definitions (though as we're discovering, it seems like this is a bit over-simplified and perhaps it'd be better if we could somehow share the type definitions from support-dotcom-components).
* a thin wrapper around `fetch` for the end of article api call (`getBodyEnd`)
* some shared libs around views retrieving and storing message views (stored in local storage)

So it seems like if we wanted we could move towards embracing `automat-client` more. For example `getBanner` which is the equivalent of `getBodyEnd` is currently implemented directly in `ReaderRevenueBanner`. Perhaps we want to move that to the client. But then if that's not been a pain point, maybe `automat-client` isn't pulling its weight in terms of usefulness to warrant having this additional lib.

I do think treating the type definitions in support-dotcom-components is the way to go. Perhaps there's a way to achieve that with a monorepo approach, where the support-dotcom-components repo also includes an NPM lib (which could replace automat-client). It seems like there might be less friction that way.